src/procedures/ControllerRedeem.ts

```diff
@@ -67,10 +67,21 @@ export class ControllerRedeem extends Procedure<ControllerRedeemProcedureArgs> {
      */
     await this.addTransaction(securityToken.controllerRedeem, {
       tag: PolyTransactionTag.ControllerRedeem,
-      resolvers: [createControllerRedeemResolver(factories, symbol, from)],
-    })({ from, value: amount, data, operatorData: log });
+      resolvers: [
+        createControllerRedeemResolver(factories, symbol, from),
+        refreshSecurityTokenFactoryResolver(factories, symbol),
+      ],
+    })({ from, value: amount, data, operatorData: reason });
   }
 }
+
+export const refreshSecurityTokenFactoryResolver = (
```

Comment:
Should be renamed to `createRefreshSecurityTokenResolver`, and the other one to `createRefreshShareholdersRresolver`